### ISSUE TITLE
[java] LawOfDemeter: false negative of (method chain call) with AnonymousClass in parameter

### ISSUE THREAD

**Issue by mohui1999** [pmd/pmd#4419]
**Affects PMD Version:** 6.55.0

**Rule:** [LawOfDemeter](https://pmd.github.io/latest/pmd_rules_java_design.html#lawofdemeter)

**Description:**
`Executors.newSingleThreadExecutor().submit()` is a method chain call. But PMD does not report it. So this may be a false negative.

**Code Sample demonstrating the issue:**
Maybe a false negative:

```java
import java.util.concurrent.Executors;
public class ExecutorsUsage {
    public void static_usage_of_Executors() {
         Executors.newSingleThreadExecutor().submit(new Runnable() {         // FN, should report here
             @Override
             public void run() {
             }
        });
    }
}
```

```java
import java.util.concurrent.Executors;
public class ExecutorsUsage {
    public void static_usage_of_Executors() {
        Executors.newSingleThreadExecutor().submit(() -> {           // here report method chain call
        });
    }
}
```

**Expected outcome:**

PMD should report a violation at line 4, but doesn't. This is a false negative.

**Running PMD through:** *[CLI]*

**Comment by mohui1999** [comment]
Fixed in PMD 7.0
